packages/applicant/src/pages/mandatory-questions/[mandatoryQuestionId]/getServerSideProps.tsx

```diff
@@ -10,14 +10,54 @@ import callServiceMethod from '../../../utils/callServiceMethod';
 import { getJwtFromCookies } from '../../../utils/jwt';
 import { routes } from '../../../utils/routes';
 
+const buildBackButtonMapper = (orgType, mandatoryQuestionId, schemeId) => {
+  const isIndividualOrNonLimitedCompany = [
+    'I am applying as an individual',
+    'Non-limited company',
+  ].includes(orgType);
+  const externalApplicationPage =
+    routes.mandatoryQuestions.externalApplicationPage(mandatoryQuestionId);
+  const summaryPage =
+    routes.mandatoryQuestions.summaryPage(mandatoryQuestionId);
+  const fundingLocationPage =
+    routes.mandatoryQuestions.fundingLocationPage(mandatoryQuestionId);
+  const fundingAmountPage =
+    routes.mandatoryQuestions.fundingAmountPage(mandatoryQuestionId);
+  const charityCommissionNumberPage =
+    routes.mandatoryQuestions.charityCommissionNumberPage(mandatoryQuestionId);
+  const companiesHouseNumberPage =
+    routes.mandatoryQuestions.companiesHouseNumberPage(mandatoryQuestionId);
+  const addressPage =
+    routes.mandatoryQuestions.addressPage(mandatoryQuestionId);
+  const namePage = routes.mandatoryQuestions.namePage(mandatoryQuestionId);
+  const typePage = routes.mandatoryQuestions.typePage(mandatoryQuestionId);
+  const startPage = routes.mandatoryQuestions.startPage(schemeId.toString());
+  return {
+    [externalApplicationPage]: summaryPage,
+    [summaryPage]: fundingLocationPage,
+    [fundingLocationPage]: fundingAmountPage,
+    ...(isIndividualOrNonLimitedCompany
+      ? {
+          [fundingAmountPage]: addressPage,
+        }
+      : {
+          [fundingAmountPage]: charityCommissionNumberPage,
+          [charityCommissionNumberPage]: companiesHouseNumberPage,
+          [companiesHouseNumberPage]: addressPage,
+        }),
```

Comment:
Stick this in a function?

Reply:
think it might just make it harder to read tbh